### PR TITLE
[Test] Unify tests for different backends

### PR DESCRIPTION
**Pull request by BarclayII**
## Description
Unified most of the computation tests into a single directory.

Added a unit-test-specific backend.

Readonly graphs are still left in the mxnet test since PyTorch doesn't support it right now.  We can remove them after we implement readonly graph with PyTorch.

A series of bugfixes related to MXNet are also included.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change

### CODE REVIEW

File: tests/compute/test_basics.py [hunk 1]
```diff
@@ -48,22 +49,22 @@ def generate_graph(grad=False):
 
 def test_batch_setter_getter():
     def _pfc(x):
-        return list(x.numpy()[:,0])
+        return list(F.zerocopy_to_numpy(x)[:,0])
```
Comment (jermainewang) [inline tests/compute/test_basics.py]: Is `[:,0]` the same for both framework?

**Comment by jermainewang** [comment]
@BarclayII would you please change the new utests from #331 too?

File: tests/compute/test_basics.py (hunk 1 above)
Comment (BarclayII) [reply tests/compute/test_basics.py]: Yes

**Comment by BarclayII** [comment]
I removed the gradient checking part in reverse graph with sharing, since overwriting a gradient-attached tensor doesn't look natural.

**Comment by jermainewang** [comment]
Once #336 is merged, help me finish the section about unittest in contribution guide?

File: CONTRIBUTORS.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ Contribution is always welcomed. A good starting place is the roadmap issue, whe
 you can find our current milestones. All contributions must go through pull requests
 and be reviewed by the committors. See our [contribution guide](https://docs.dgl.ai/contribute.html) for more details.
 
-Once your contribution is accepted and merged, congratulation, you are now an contributor to the DGL project.
+Once your contribution is accepted and merged, congratulations, you are now an contributor to the DGL project.
```
Comment (szha) [inline CONTRIBUTORS.md]: a contributor

**Pull request merged**